lib/vmpooler/pool_manager.rb

```diff
@@ -295,22 +296,46 @@ def generate_and_check_hostname(_pool_name)
     end
 
     def find_unique_hostname(pool_name)
+      # generate hostname that is not already in use in vmpooler
+      # also check that no dns record already exists
       hostname_retries = 0
       max_hostname_retries = 3
       while hostname_retries < max_hostname_retries
         hostname, available = generate_and_check_hostname(pool_name)
-        break if available
+        domain = $config[:config]['domain']
+        dns_ip, dns_available = check_dns_available(hostname, domain)
+        break if available && dns_available
```

Comment:
with the addition of the dns availability check, it might be good to rename `available` to `hostname_available` to differentiate what each of these variables check for